src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java

```diff
@@ -251,6 +252,30 @@ private void checkCommErrors() throws ServerException {
 		}
 	}
 
+	/**
+	 * Returns true if connected with a recent event.
+	 * If binlog heartbeats are disabled, just returns
+	 * whether there is a connection.
+	 */
+	private boolean isConnectionAlive() {
+		if (!isConnected) {
+			return false;
+		}
+		boolean isAlive = true;
+		if (client.getHeartbeatInterval() > 0L) {
```

Comment:
would be nice to check `enableHeartbeat` here instead.

Reply:
It didn't seem worth adding a permanent reference to `mysqlConfig` which is otherwise only available in the constructor. But see below, I've changed it now.